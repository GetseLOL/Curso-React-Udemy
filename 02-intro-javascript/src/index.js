// async - await

// const getImagesPromesa = () =>
//   new Promise((resolve) => resolve("https://google.com"));

const getImagesPromesa = async () => {
  try {
    return "https://google.com";
  } catch (error) {
    console.error(error);
  }
};

getImagesPromesa().then(console.log);
