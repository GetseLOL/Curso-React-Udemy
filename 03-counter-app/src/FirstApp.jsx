import React from "react";
import PropTypes from "prop-types";
import "./FirstApp.css";

const FirstApp = ({ title, subTitle, name }) => {
  return (
    <>
      <h1 data-testid="test-title">{title}</h1>
      <h2>{subTitle}</h2>
      <h2>{subTitle}</h2>
      <h3>{name}</h3>
    </>
  );
};

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
};

FirstApp.defaultProps = {
  // title: "No hay titulo",
  subTitle: "No hay subtitulo",
  name: "No hay titulo",
};

export default FirstApp;
