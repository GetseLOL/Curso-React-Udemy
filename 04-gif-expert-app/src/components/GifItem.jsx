import PropTypes from "prop-types";

export const GifItem = ({ title, url, id }) => {
  return (
    <div className="card">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
};

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

/**
 * Tarea
 * 1. AÃ±adir props
 *  a. Titulo obligatorio
 *  b. URL obligatorio
 *
 * 2. Evaluar snapshot
 */
