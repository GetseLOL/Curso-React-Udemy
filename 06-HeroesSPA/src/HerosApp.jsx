import React from "react";
import { AppRouter } from "./router/AppRouter";
import { AuthProvider } from "./context/AuthProvider";

export const HerosApp = () => {
  return (
    <AuthProvider>
      <AppRouter />
    </AuthProvider>
  );
};
