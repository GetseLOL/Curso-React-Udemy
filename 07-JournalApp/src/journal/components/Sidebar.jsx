import { TurnedInNot } from "@mui/icons-material";
import {
  Drawer,
  Box,
  Toolbar,
  Typography,
  Divider,
  List,
  ListItem,
  ListItemButton,
  ListItemIcon,
  Grid,
  ListItemText,
} from "@mui/material";

export const Sidebar = ({ drawerWidth = 240 }) => {
  return (
    <Box
      component="nav"
      sx={{
        width: { sm: drawerWidth },
        flexShrink: { sm: 0 },
      }}
    >
      <Drawer
        variant="permanent"
        open
        sx={{
          display: {
            xs: "block",
            "& .MuiDrawer-paper": {
              boxSizing: "border-box",
              width: drawerWidth,
            },
          },
        }}
      >
        <Toolbar>
          <Typography variant="h6" noWrap component="div">
            Jhon Wichik
          </Typography>
        </Toolbar>
        <Divider />
        <List>
          {["Enero", "Febrero", "Marzo", "Abril"].map((text) => (
            <ListItem key={text} disablePadding>
              <ListItemButton>
                <ListItemIcon>
                  <TurnedInNot />
                </ListItemIcon>
                <Grid container>
                  <ListItemText primary={text} />
                  <ListItemText
                    secondary={
                      "Lorem ipsum dolor sit amet consectetur adipisicing elit. Debitis deserunt."
                    }
                  />
                </Grid>
              </ListItemButton>
            </ListItem>
          ))}
        </List>
      </Drawer>
    </Box>
  );
};
