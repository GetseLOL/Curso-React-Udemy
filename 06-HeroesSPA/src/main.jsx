import React from "react";
import ReactDOM from "react-dom/client";

import { BrowserRouter } from "react-router-dom";

import "./styles.css";
import { HerosApp } from "./HerosApp";

ReactDOM.createRoot(document.getElementById("root")).render(
  <BrowserRouter>
    {/* <React.StrictMode> */}
      <HerosApp />
    {/* </React.StrictMode> */}
  </BrowserRouter>
);
