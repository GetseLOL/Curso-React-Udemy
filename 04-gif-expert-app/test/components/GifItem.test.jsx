import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components";

describe("Pruebas en <GifItem />", () => {
  const title = "Iron Cat";
  const url =
    "https://i.kym-cdn.com/photos/images/newsfeed/002/164/868/943.jpg";

  test("Debe hacer match con el snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("Debe de mostrar la imagen con el URL y el ALT indicado", () => {
    render(<GifItem title={title} url={url} />);
    // screen.debug()
    // console.log(screen.getByRole("img").alt);
    // expect(screen.getByRole("img").alt).toBe(title);
    // expect(screen.getByRole("img").src).toBe(url);
    const { alt, src } = screen.getByRole("img");
    expect(alt).toBe(title);
    expect(src).toBe(url);
  });


  test('Debe mostrar el titulo en el component ', () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  })
  
});
