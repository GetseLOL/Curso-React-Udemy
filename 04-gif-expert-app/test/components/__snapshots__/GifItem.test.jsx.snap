// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Iron Cat"
      src="https://i.kym-cdn.com/photos/images/newsfeed/002/164/868/943.jpg"
    />
    <p>
      Iron Cat
    </p>
  </div>
</div>
`;
